oneview_redfish_toolkit/api/zone.py

```diff
@@ -0,0 +1,98 @@
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from oneview_redfish_toolkit.api.computer_system import ComputerSystem
+from oneview_redfish_toolkit.api.redfish_json_validator \
+    import RedfishJsonValidator
+from oneview_redfish_toolkit.api.resource_block_collection import \
+    ResourceBlockCollection
+from oneview_redfish_toolkit.api import status_mapping
+from oneview_redfish_toolkit.api.zone_collection import ZoneCollection
+
+
+class Zone(RedfishJsonValidator):
+    """Creates a Zone Redfish dict
+
+        Populates self.redfish with Zone data retrieved from
+        OneView
+    """
+
+    SCHEMA_NAME = 'Zone'
+
+    def __init__(self, profile_template, available_targets_obj):
+        """Zone constructor
+
+            Populates self.redfish with the contents of
+            server profile template dict from Oneview
+
+            Args:
+                profile_template: Oneview's server profile template dict
+        """
+        super().__init__(self.SCHEMA_NAME)
+
+        available_targets = available_targets_obj["targets"]
+
+        self.redfish["@odata.type"] = "#Zone.v1_1_0.Zone"
+        self.redfish["Id"] = profile_template["uri"].split("/")[-1]
+        self.redfish["Name"] = "Resource Zone " + profile_template["name"]
+        self.redfish["Status"] = status_mapping.STATUS_MAP["OK"]
+
+        self.redfish["Links"] = dict()
+        self.redfish["Links"]["ResourceBlocks"] = list()
+
+        self.fill_resource_blocks(profile_template, available_targets)
+
+        self.capabilities_key = "@Redfish.CollectionCapabilities"
+        self.redfish[self.capabilities_key] = dict()
+        self.redfish[self.capabilities_key]["@odata.type"] = \
+            "#CollectionCapabilities.v1_0_0.CollectionCapabilities"
+        self.redfish[self.capabilities_key]["Capabilities"] = list()
+
+        self.fill_capabilities_collection()
+
+        self.redfish["@odata.context"] = "/redfish/v1/$metadata#Zone.Zone"
+        self.redfish["@odata.id"] = ZoneCollection.BASE_URI + "/" +\
+            self.redfish["Id"]
+
+        self._validate()
+
+    def fill_resource_blocks(self, profile_template, available_targets):
+        for item in available_targets:
+            self.add_resource_block_item_to_links(item, "serverHardwareUri")
+
+        self.add_resource_block_item_to_links(profile_template, "uri")
+
+    def add_resource_block_item_to_links(self, original_dict, uri_key):
+        uuid = original_dict[uri_key].split("/")[-1]
+        dict_item = dict()
+        dict_item["@odata.id"] = ResourceBlockCollection.BASE_URI + "/" + uuid
+        self.redfish["Links"]["ResourceBlocks"].append(dict_item)
+
+    def fill_capabilities_collection(self):
+        capability = {
+            "CapabilitiesObject": {
+                "@odata.id": "/redfish/v1/Systems/Capabilities"
+                # todo update to use a constant when the
```

Comment:
@data.id should have SPT UUID so we know how to build the Capabilities object, we can update that when adding the other resource blocks